### DISCUSSION TITLE
question about multiple server group

### DISCUSSION THREAD

**Discussion by ravibhooshan** [jacksontj/promxy#529] [Q&A]
Hi,
Wish you all a very happy new year.

I have below questions, can someone please help me understand this.

- If I define multiple servergroup with 2 server server in each group. Each server group represent one independent  prometheus setup. 

So if I query  data from promxy, will that query run on all 4 nodes. If yes, how can I restrict queires to go a particular servergroup only. 

Thanks,
Ravi

**Comment by jacksontj** [comment]
The query will go to all the nodes required to fulfill the query. The simplest way to restrict the query is to add a `labels` section to the servergroup config and ensure that matcher exists in the query being sent. For example if you have `servergroup="a"` as a matcher and the servergroups are `a` and `b` promxy will only send that query to the nodes configured with `a`.

But fundamentally promxy is limited to the knowledge it has of the downstream servergroups. In the event of `labels` we configured them in. Promxy also supports `label_filter` (https://github.com/jacksontj/promxy/blob/master/pkg/servergroup/config.go#L164-L188) which will dynamically pull some label information from downstreams in an attempt to reduce calls made.

Hopefully that answers the question?
